File: python/sample2.py
# -*- coding: utf-8 -*-

from time import time
import numpy as np
from copy import deepcopy

start = time()
a = np.arange(25).reshape(5, 5)
b = deepcopy(a) * 2

print(np.einsum("ii", a))
print(np.einsum("ii->i", a))
print(np.einsum("ij->i", a))
print(np.einsum("ij->j", a))
print(np.einsum("ij->ji", a))

print(np.einsum("ij,jk->ik", a, b))
print(np.einsum("ij,jk->ki", a, b))
print(np.einsum("ij,ij->ij", a, b))

print(time() - start)

File: julia/sample2.jl
# -*- coding: utf-8 -*-

using Einsum

tic()
a = transpose(reshape(0:24, (5, 5)))
b = copy(a) * 2

A = Array{Int64}(size(a)[1])
@einsimd A[i] = a[i, i]
println(A)

@einsimd A[i] = a[i, j]
println(A)

@einsimd A[j] = a[i, j]
println(A)

AA = Array{Int64}(size(a))
@einsimd AA[i, j] = a[j, i]
println(AA)

AB = Array{Int64}(size(a))
@einsimd AB[i, k] = a[i, j] * b[j, k]
println(AB)

@einsimd AB[k, i] = a[i, j] * b[j, k]
println(AB)

@einsimd AB[i, j] = a[i, j] * b[i, j]
println(AB)

println(toq())
